main/src/ca/uwaterloo/flix/language/phase/Weeder.scala

```diff
@@ -1603,34 +1603,42 @@ object Weeder extends Phase[ParsedAst.Program, WeededAst.Program] {
     case ParsedAst.Type.Tuple(sp1, elms, sp2) => WeededAst.Type.Tuple(elms.toList.map(visitType), mkSL(sp1, sp2))
 
     case ParsedAst.Type.Record(sp1, fields, restOpt, sp2) =>
-      val init = restOpt match {
-        case None => WeededAst.Type.RecordEmpty(mkSL(sp1, sp2))
-        case Some(base) => WeededAst.Type.Var(base, mkSL(sp1, sp2))
-      }
-      fields.foldRight(init: WeededAst.Type) {
-        case (ParsedAst.RecordFieldType(ssp1, l, t, ssp2), acc) =>
-          WeededAst.Type.RecordExtend(l, visitType(t), acc, mkSL(ssp1, ssp2))
+      if (fields.isEmpty && restOpt.isDefined) {
```

Comment:
I really prefer pattern matches:

```
(fields, restOpt) match 
  case (Nil, Some(...)) => 
  case (...) => 
```

Reply:
And then add some comments:

// Case 1: The type is of the form { | a } with no record fields. We know `a` is a record type variable.
// Case 2: The type is ...

or similar :)

Reply:
Yeah I felt yucky making those if-statements. I've cleaned it up.